### ISSUE TITLE
test user_agent_tls.t - missing ca in first test without certificate

### ISSUE THREAD

**Issue by pheuch** [mojolicious/mojo#1267]
* Mojolicious version: 8.02
* Perl version: 5.20.1
* Operating system: AIX 7.2

### Steps to reproduce the behavior
In test t/mojo/user_agent_tls.t  the first test without a certificate does not set the ca, which seems to be required in this mojo version. After setting the ca the test does fail

Change in t/mojo/user_agent_tls.t - added line 38

>  +36  # No certificate
>  +37  my $ua = Mojo::UserAgent->new(ioloop => Mojo::IOLoop->singleton);
>  +38  $ua->ca('t/mojo/certs/ca.crt');  # <= new line
>  +39  my $tx = $ua->get("https://127.0.0.1:$port");
> +40  ok $tx->error, 'has error';+36  # No certificate
   

### Expected behavior
The server should not accept the connections The test "has error" should me true

### Actual behavior
With the ca given, the server does accept the connection. The test without the change was ok, because the client die reject the connection. With the change the client does accept the connection and the server should reject it, because of the missing client certificate. :

> not ok 1 - has error
>    Failed test 'has error'
>   at t/mojo/user_agent_tls.t2 line 40.
> not ok 2 - has error
>   Failed test 'has error'
>   at t/mojo/user_agent_tls.t2 line 42.

**Comment by kraih** [comment]
Afraid i don't quite understand your issue, but i did add more TLS tests.

**Comment by pheuch** [comment]
@kraih Sorry to be not clear enough. In the original test on line 36-41 these is the a duplicate test. 
>36 # No certificate
> 37 my $ua = Mojo::UserAgent->new(ioloop => Mojo::IOLoop->singleton);
> 38 my $tx = $ua->get("https://127.0.0.1:$port");
> 39 ok $tx->error, 'has error';
> 40 $tx = $ua->get("https://127.0.0.1:$port");
> 41 ok $tx->error, 'has error';

Lines 38 and 40 are doing the same. I think it should test  "get an error because the ca option of user agent is missing".  And the next should be, set ca and get an error because the client certificate is missing.  So I added the line $ua->ca('t/mojo/certs/ca.crt');  (should be inserted after line 39  and not after line 37 as I wrote at the beginning).
But in this case, no error is reported. The connection works. This is a bit unexpected for me.

Ok, just noticed that in  my old version (7.72) I know  compared  to the current version the default for verify did change. In the old version  in Mojo::IOLoop::TLS  SSL_verify_mode was set to 0x03 if ca was specified.  In current version the default of verify is 0x00 and is independent of the setting of CA.
Is this intended?  Documentation is a bit unclear for this (at least in Mojo::Server::Daemon).
